### PR TITLE
[Android] prevent checkbox from being checked for ERC721 in registry

### PR DESCRIPTION
**Pull request by wchen342**
Resolves https://github.com/brave/brave-browser/issues/27114

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/crypto_wallet/adapters/WalletCoinAdapter.java [hunk 1]
```diff
@@ -158,6 +158,10 @@ public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
                             @Override
                             public void onCheckedChanged(
                                     CompoundButton buttonView, boolean isChecked) {
+                                if  ("noOnClickListener".equals((String) holder.assetCheck.getTag())) {
+                                    holder.assetCheck.setTag(null);
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/crypto_wallet/adapters/WalletCoinAdapter.java:162]: I don't think I understand that part. It looks like we skip the whole block currently if we uncheck the checkbox after it's been checked? We still need to do the loop below in that case. Why we cannot just check if we unchecking and don't show the add token dialog in that case only?

Comment (wchen342) [reply android/java/org/chromium/chrome/browser/crypto_wallet/adapters/WalletCoinAdapter.java:162]: I think I see what you meant about the loop, yes the check should be put below the loop, I will correct that.

Comment (wchen342) [reply android/java/org/chromium/chrome/browser/crypto_wallet/adapters/WalletCoinAdapter.java]: Updated now.

**Comment by srirambv** [comment]
Verification passed on Oppo Reno 5 with Android 12 running 1.48.32 x64 nightly build
- Verified steps from issue description & PR
- Verified add token form is only shown when the token is clicked 
- Verified the token doesn't add a check mark and only brings up the modal every time the token is clicked

https://user-images.githubusercontent.com/17010094/206499010-01c94fae-0c01-4209-9b86-0139b590d5f4.mp4

**Pull request merged**